### PR TITLE
Fix image name for example webhook, and other nitpicks

### PR DESCRIPTION
**Pull request by bswartz**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
The image name for the webhook was wrong in the example.
Also fixed a few nitpicks

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: deploy/kubernetes/webhook-example/webhook.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ spec:
     spec:
       containers:
       - name: snapshot-validation
-        image: k8s.gcr.io/sig-storage/validation-webhook:v2.2.0 # change the image if you wish to use your own custom validation server image
+        image: k8s.gcr.io/sig-storage/snapshot-validation-webhook:v2.2.0 # change the image if you wish to use your own custom validation server image
```
Comment (xing-yang) [inline deploy/kubernetes/webhook-example/webhook.yaml:20]: This should be v3.0.0.
We didn't cut v2.2.0.  We cut v3.3.0.

**Review by xing-yang — COMMENTED** [review]
Can you also update the images under external-snapshotter/deploy?
Images should be updated to the following:
```
k8s.gcr.io/sig-storage/snapshot-controller:v3.0.0
k8s.gcr.io/sig-storage/csi-snapshotter:v3.0.0
```

**Comment by xing-yang** [comment]
We should update this one too:

https://github.com/kubernetes-csi/external-snapshotter/blob/master/deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml#L75
New image:
k8s.gcr.io/sig-storage/csi-provisioner:v2.0.2

We also need to update the hostpath driver when the new image is cut:
https://github.com/kubernetes-csi/external-snapshotter/blob/master/deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml#L100

File: deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml [hunk 2]
```diff
@@ -72,7 +72,7 @@ spec:
       serviceAccount: csi-snapshotter
       containers:
         - name: csi-provisioner
-          image: quay.io/k8scsi/csi-provisioner:v1.5.0
+          image: k8s.gcr.io/sig-storage/csi-provisioner:v1.5.0
```
Comment (xing-yang) [inline deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml:75]: The latest is v2.0.2

File: deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml [hunk 3]
```diff
@@ -97,7 +97,7 @@ spec:
             - name: socket-dir
               mountPath: /csi
         - name: hostpath
-          image: quay.io/k8scsi/hostpathplugin:v1.2.0
+          image: k8s.gcr.io/sig-storage/hostpathplugin:v1.2.0
```
Comment (xing-yang) [inline deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml:100]: This image does not exist.  Let's wait for the new csi hostpath driver image to be cut.

Comment (yuxiangqian) [reply deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml]: all images will go to k8s.gcr.io in the future?

Comment (msau42) [reply deploy/kubernetes/csi-snapshotter/setup-csi-snapshotter.yaml]: You can use hostpath driver 1.4.0

**Comment by msau42** [comment]
@xing-yang longer term, do you see a way to consolidate all these image examples so we don't have to remember to update all these places every time we cut new releases? I am thinking things like making the image version templatable and using git tags or something to get the latest released version.

**Pull request merged**